addons/web/static/tests/views/form/form_view_tests.js

```diff
@@ -4021,6 +4021,33 @@ QUnit.module("Views", (hooks) => {
         assert.containsNone(target, ".o_cp_action_menus span:contains(Unarchive)");
     });
 
+    QUnit.test("archive action not shown with readonly active field", async function (assert) {
+
+        // add active field on partner model, but do not put it in the view
+        serverData.models.partner.fields.active = {
+            string: "Active",
+            type: "char",
+            default: true,
+            readonly: true,
+        };
+        await makeView({
+            type: "form",
+            resModel: "partner",
+            resId: 1,
+            serverData,
+            arch: `<form><field name="foo"/></form>`,
+            actionMenus: {},
+        });
+        await click(target, ".o_cp_action_menus .dropdown-toggle");
+        assert.deepEqual(
+            [...target.querySelectorAll(".o_menu_item")].map((el) => el.textContent),
+         ["Duplicate", "Delete"],
```

Comment:
Please make sure you apply the linting rules provided with the JS tooling: https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines#tooling